etl/steps/data/meadow/un_sdg/2022-05-26/un_sdg.py

```diff
@@ -0,0 +1,404 @@
+    # load significantly faster
+    df = pd.read_csv(local_file, low_memory=False)
+
+    if query:
+        df = df.query(query)
+
+    log.info("un_sdg.load_and_clean")
+    df = load_and_clean(df)
+    log.info("Size of dataframe", rows=df.shape[0], colums=df.shape[1])
+    log.info("un_sdg.create_dataframe")
+    full_df = create_dataframe(df)
+    full_df.columns = [underscore(c) for c in full_df.columns]
+    full_df = full_df[
+        [
+            "country",
+            "year",
+            "variable_name",
+            "source",
+            "value",
+            "units_long",
+            "short_unit",
+        ]
+    ]
+    # verify_integrity checks for duplicates
+    log.info("Size of dataframe", rows=full_df.shape[0], colums=full_df.shape[1])
+
+    assert full_df["country"].notnull().all()
+    assert full_df["variable_name"].notnull().all()
+    assert (
+        full_df[
+            (
+                full_df[["country", "year", "variable_name", "value"]].duplicated(
+                    keep=False
+                )
+            )
+        ].shape[0]
+        == 0
+    ), "Unexpected duplicates in dataframe."
+    assert (
+        not full_df.isnull().all(axis=1).any()
+    ), "Unexpected state: One or more rows contains only NaN values."
+
+    # Create a primary key index
+    full_df = full_df.set_index(
+        ["country", "year", "variable_name"], verify_integrity=True
+    )
+
+    # creates the dataset and adds a table
+    ds = Dataset.create_empty(dest_dir)
+    ds.metadata = DatasetMeta(
+        short_name=walden_ds.short_name,
+        title=walden_ds.name,
+        namespace=walden_ds.namespace,
+        description=walden_ds.description,
+    )
+    tb = Table(full_df)
+    tb.metadata = TableMeta(
+        short_name=Path(__file__).stem,
+        title=walden_ds.name,
+        description=walden_ds.description,
+        primary_key=list(full_df.index.names),
+    )
+    ds.add(tb)
+    ds.save()
+    log.info("un_sdg.end")
+
+
+def create_dataframe(original_df: pd.DataFrame) -> None:
+    # Removing the square brackets from the indicator column
+    original_df = original_df.copy(deep=False)
+
+    original_df = original_df.rename(columns=lambda k: re.sub(r"[\[\]]", "", k))  # type: ignore
+
+    unit_description = attributes_description()
+
+    dim_description = dimensions_description()
+
+    original_df["Units_long"] = original_df["Units"].map(unit_description)
+    original_df["short_unit"] = create_short_unit(original_df["Units_long"])
+
+    original_df = manual_clean_data(original_df)
+
+    init_dimensions = tuple(dim_description["id"].unique())
+    init_non_dimensions = tuple(
+        [c for c in original_df.columns if c not in set(init_dimensions)]
+    )
+    all_series = (
+        original_df[["Indicator", "SeriesCode"]].drop_duplicates().reset_index()
+    )
+
+    all_series = original_df.groupby(["Indicator", "SeriesCode"])
```

Comment:
Shouldn't this be
```
all_series = all_series.groupby(["Indicator", "SeriesCode"])
```
you should also use a different name for the grouped object, otherwise type checker is going to complain.

Reply:
Ah yes this is a mistake

`all_series = (
        original_df[["Indicator", "SeriesCode"]].drop_duplicates().reset_index()
    )`
shouldn't be in there, will delete!